tiatoolbox/models/architecture/vanilla.py

```diff
@@ -93,43 +107,83 @@ def _get_timm_architecture(
             Whether to load pretrained weights.
 
     Returns:
-        A ready-to-use timm model.
+        list[nn.Sequential, ...] | nn.Sequential:
+            A ready-to-use timm model.
 
     Raises:
         ValueError:
             If the backbone architecture is not supported.
 
+    Example:
+        >>> model = _get_timm_architecture("UNI", pretrained=True)
+        >>> print(model)
+    
     """
     if arch_name in [f"efficientnet_b{i}" for i in range(8)]:
         model = timm.create_model(arch_name, pretrained=pretrained)
         return nn.Sequential(*list(model.children())[:-1])
 
-    if arch_name == "UNI":  # pragma: no cover
+    arch_map = {
         # UNI tile encoder: https://huggingface.co/MahmoodLab/UNI
-        # Coverage skipped timm API is tested using efficient U-Net.
-        return timm.create_model(
-            "hf-hub:MahmoodLab/UNI",
-            pretrained=pretrained,
-            init_values=1e-5,
-            dynamic_img_size=True,
-        )
-
-    if arch_name == "prov-gigapath":  # pragma: no cover
+        "UNI": {
+            "model": "hf-hub:MahmoodLab/UNI",
+            "init_values": 1e-5,
+            "dynamic_img_size": True,
+        },
         # Prov-GigaPath tile encoder: https://huggingface.co/prov-gigapath/prov-gigapath
-        # Coverage skipped timm API is tested using efficient U-Net.
-        return timm.create_model(
-            "hf_hub:prov-gigapath/prov-gigapath",
-            pretrained=pretrained,
-        )
-
-    if arch_name == "H-optimus-0":  # pragma: no cover
+        "prov-gigapath": {"model": "hf_hub:prov-gigapath/prov-gigapath"},
         # H-Optimus-0 tile encoder: https://huggingface.co/bioptimus/H-optimus-0
+        "H-optimus-0": {
+            "model": "hf-hub:bioptimus/H-optimus-0",
+            "init_values": 1e-5,
+            "dynamic_img_size": False,
+        },
+        # UNI2-h tile encoder: https://huggingface.co/MahmoodLab/UNI2-h
+        "UNI2": {
```

Comment:
Please can you highlight this in the PR description that now foundation models include additional parameters from the model?

Reply:
Ofcourse!